core/src/main/java/org/geysermc/geyser/translator/protocol/bedrock/entity/player/BedrockSetPlayerGameTypeTranslator.java

```diff
@@ -25,23 +25,41 @@
 
 package org.geysermc.geyser.translator.protocol.bedrock.entity.player;
 
+import com.github.steveice10.mc.protocol.data.game.entity.player.GameMode;
 import org.cloudburstmc.protocol.bedrock.packet.SetPlayerGameTypePacket;
 import org.geysermc.geyser.session.GeyserSession;
 import org.geysermc.geyser.translator.protocol.PacketTranslator;
 import org.geysermc.geyser.translator.protocol.Translator;
 
 /**
  * In vanilla Bedrock, if you have operator status, this sets the player's gamemode without confirmation from the server.
- * Since we have a custom server option to request the gamemode, we just reset the gamemode and ignore this.
+ * With operator status, the Gamemode change is sent to the Java server, if it is not present, the gamemode is not changed.
  */
 @Translator(packet = SetPlayerGameTypePacket.class)
 public class BedrockSetPlayerGameTypeTranslator extends PacketTranslator<SetPlayerGameTypePacket> {
 
+    /**
+     * Sets client game mode for the server via the Bedrock client's "world" menu (given sufficient permissions).
+     */
     @Override
     public void translate(GeyserSession session, SetPlayerGameTypePacket packet) {
-        // no
-        SetPlayerGameTypePacket playerGameTypePacket = new SetPlayerGameTypePacket();
-        playerGameTypePacket.setGamemode(session.getGameMode().ordinal());
-        session.sendUpstreamPacket(playerGameTypePacket);
+        // yes, if you are OP
+        if (session.getOpPermissionLevel() >= 2 || session.hasPermission("geyser.settings.server")) {
+            if (packet.getGamemode() != session.getGameMode().ordinal()) {
+                // For some reason, spectator mode is... 6?
+                GameMode gameMode = switch (packet.getGamemode()) {
+                    case 1 -> GameMode.CREATIVE;
+                    case 2 -> GameMode.ADVENTURE;
+                    case 5 -> session.getGeyser().getWorldManager().getDefaultGameMode(session);
```

Comment:
is `5` an unassigned/undefined value or?

Reply:
Bedrock has a few [fun valid Gamemodes](https://github.com/CloudburstMC/Protocol/blob/81ef6bbb3bd5acba3de6dbe987bc41731f90b8c7/bedrock-codec/src/main/java/org/cloudburstmc/protocol/bedrock/data/GameType.java#L11C11-L11C11); such as Gamemode 3/4, or here, 5 "custom"
Which for BDS basically means that the Bedrock client will automatically (without even informing the server, because Bedrock ™️) switch its Gamemode to the default Gamemode. Since that is not default Java behavior (unless force-gamemode is enabled), we just send the new Gamemode (current default gamemode) once and prohibit automatic client gamemode changes when the default gamemode is changed [here](https://github.com/GeyserMC/Geyser/pull/4062/files#diff-81ce176bf7aa41589639ba8e48368573ef6baefaddff6d803641063c45f7a4c2R46-R50)